rclcpp_action/test/test_server_goal_handle.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/exceptions.hpp>
+#include <rclcpp/node.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <test_msgs/action/fibonacci.hpp>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <vector>
+
+#include "action_msgs/msg/goal_info.h"
+#include "rclcpp_action/server_goal_handle.hpp"
+#include "./mocking_utils/patch.hpp"
+
+class FibonacciServerGoalHandle
+  : public rclcpp_action::ServerGoalHandle<test_msgs::action::Fibonacci>
+{
+public:
+  FibonacciServerGoalHandle(
+    std::shared_ptr<rcl_action_goal_handle_t> rcl_handle,
+    rclcpp_action::GoalUUID uuid,
+    std::shared_ptr<const test_msgs::action::Fibonacci::Goal> goal,
+    std::function<void(
+      const rclcpp_action::GoalUUID &, std::shared_ptr<void>)> on_terminal_state,
+    std::function<void(const rclcpp_action::GoalUUID &)> on_executing,
+    std::function<void(
+      std::shared_ptr<test_msgs::action::Fibonacci::Impl::FeedbackMessage>)> publish_feedback)
+  : rclcpp_action::ServerGoalHandle<test_msgs::action::Fibonacci>(
+      rcl_handle, uuid, goal, on_terminal_state, on_executing, publish_feedback) {}
```

Comment:
@brawner nit: you can "use" base class constructors too (see [here](https://en.cppreference.com/w/cpp/language/using_declaration#Inheriting_constructors)).

Reply:
Argh, almost :)